rustplus/api/remote/rustws.py

```diff
@@ -79,6 +81,15 @@ async def connect(
                     address += f"?v={str(self.magic_value)}"
                     self.connection = await connect(address, close_timeout=0, ping_interval=None)
                     self.connected_time = time.time()
+
+                    if asyncio.iscoroutinefunction(self.on_success):
+                        await self.on_success()
+                    else:
+                        try:
+                            self.on_success()
```

Comment:
I know what you said about needing this try/except for if one isn’t provided, but I think it would be better to instead just check before any of the other logic whether or not it is None. Then we can avoid the gross exception. Leave the try, except in thought but change the TypeError to Exception and then log the error. This just catches any stupid mistakes in the callback that could happen.